app/services/team_move_service.rb

```diff
@@ -103,8 +103,12 @@ def restore_new_team_name_to_original_name
   end
 
   def restore_users_for_old_team
-    @keep_user_roles.each do |ur|
-      TeamsUsersRole.create!(team: Team.find(ur[0]), user: User.find(ur[1]), role: ur[2])
+    @keep_user_roles.each do |team_id, user_id, role|
+      team =  Team.find_by_id(team_id)
+      user = User.find_by_id(user_id)
+      if team && user
+        TeamsUsersRole.create!(team: team, user: user, role: role)
+      end
```

Comment:
@Polsonby,  just being a little picky,  I was wondering whether we need to have a way to throw/log such errors,  but I am happy to leave it like the current way

Reply:
Great point... going to merge just now, but will have a think about this more generally - thanks!